drivers/hmis/app/graphql/types/hmis_schema/client.rb

```diff
@@ -104,5 +114,14 @@ def image
     def user
       load_ar_association(object, :user)
     end
+
+    def ssn
+      return object.ssn if current_user.can_view_full_ssn_for?(object)
+      return object&.ssn&.sub(/^.*?(\d{4})$/, 'XXXXX\1') if current_user.can_view_partial_ssn_for?(object)
```

Comment:
Does the display in a useful SSN format happen in the front-end? (eg. xxx-xx-xxxx)

Reply:
I think it only shows the last 4, but I don't remember getting any specific direction on it, so we can adjust if needed.

My thought is that it should only show the last 4 and it should indicate that those are the last four. With this change, the backend will send the valid SSNs `123456789`, `XXX126789`, and `XXXXX6789` as just `XXXXX6789`. This means that the frontend will treat those all as equal if you get the full SSN treatment, which seems wrong to me. That's why I think we should show something like "Last 4 of SSN: 6789" rather than "XXX-XX-6789". Just my $0.02

Reply:
Yep, that's what it looks like in the frontend. I like the approach.